src/module/smart-sessions/usage.ts

```diff
@@ -511,3 +513,39 @@ export const getDisableActionPolicies = ({
     }),
   }
 }
+
+export const toActionConfig = (config: ActionConfig): RawActionConfig => {
+  // Ensure we always have 16 rules, filling with default values if necessary
+  const filledRules = [...config.paramRules.rules];
+
+  // Fill the rest with default ParamRule if the length is less than 16
+  while (filledRules.length < 16) {
+    filledRules.push({
+      condition: 0,  // Default condition (EQUAL)
+      offset: 0,  // Default offsetIndex
+      isLimited: false,  // Default isLimited flag
+      ref: "0x0000000000000000000000000000000000000000000000000000000000000000",  // Default bytes32 ref
+      usage: {
+        limit: BigInt(0),  // Default limit
+        used: BigInt(0)    // Default used
+      }
+    });
+  }
+
+  return {
+    valueLimitPerUse: BigInt(config.valueLimitPerUse),
+    paramRules: {
+      length: BigInt(config.paramRules.length),
+      rules: filledRules.map((rule) => {
+        const parsedRef = parseReferenceValue(rule.ref);
+        return {
+          condition: rule.condition,
+          offset: BigInt(rule.offset) * BigInt(32),  // Ensure correct offset calculation
```

Comment:
`offset: BigInt(rule.offsetIndex) * BigInt(32)`. <- This makes it clearer what's going on.

Reply:
Maybe rename the relevant field / type to offsetIndex?

Reply:
offsetIndex var doesn't exist in the contract and it has been removed from module-sdk also.

Reply:
before there was some types such as

`export type RawActionConfig = {
  valueLimitPerUse: bigint
  paramRules: RawParamRules
}

export type RawParamRules = {
  length: bigint
  rules: RawParamRule[]
}

export type RawParamRule = {
  condition: ParamCondition
  offset: bigint
  isLimited: boolean
  ref: Hex
  usage: LimitUsage
}`

over

`export type ActionConfig = {
  valueLimitPerUse: bigint
  paramRules: {
    length: number
    rules: Rule[]
  }
}

export type Rule = {
  /**
   * EQUAL = 0,
   * GREATER_THAN = 1,
   * LESS_THAN = 2,
   * GREATER_THAN_OR_EQUAL = 3,
   * LESS_THAN_OR_EQUAL = 4,
   * NOT_EQUAL = 5
   */
  condition: ParamCondition
  /**
   * The offset in the calldata where the value to be checked is located.
   * The offset is in multiples of 32 bytes. (Note: not the offsetIndex)
   * The offsetIndex is generally the index of the arg in the method that you wish to target.
   * The exception is when the arg is in an array
   * In this case, the offsetIndex needs to be figured out using its position in the array
   * (See the 'use-of-dynamic-types' example below for how to figure out the offsetIndex for an array)
   *
   * https://docs.soliditylang.org/en/develop/abi-spec.html#use-of-dynamic-types
   *
   * */
  offsetIndex: bigint
  /**
   * If the rule is limited, the usage object will contain the limit and the used values.
   */
  isLimited: boolean
  /**
   * The reference value to compare against. You can pass in the raw hex value or a human-friendly value.
   * Use the raw hex value if you are sure of the value you are passing in.
   */
  ref: AnyReferenceValue
  /**
   * The usage object will contain the limit and the used values, and is only required if the isLimited property is true.
   */
  usage: LimitUsage
}`

and this util toActionConfig converted ActionConfig it into RawActionConfig